realm/realm-library/src/objectServer/java/io/realm/mongodb/User.java

```diff
@@ -502,8 +502,8 @@ public boolean equals(@Nullable Object o) {
 
         User user = (User) o;
 
-        if (!osUser.equals(user.osUser)) return false;
-        return app.equals(user.app);
+        if (!osUser.getIdentity().equals(user.osUser.getIdentity())) return false;
```

Comment:
Again; it would maybe be nice to document the criteria for `equals`, but at least it is derivable from the code here.